examples/text-classification/run_nlp_glue.py

```diff
@@ -0,0 +1,311 @@
+        cache_dir=model_args.cache_dir,
+        # use_fast=True,
+    )
+
+    # Download, tokenize and prepare datasets for training a PyTorch model
+    task_name = data_args.task_name.replace("-", "")  # We used to allow 'sts-b' for 'stsb'
+    glue = nlp.load_dataset("glue", name=task_name)
+
+    def tokenize(batch):
+        """ Tokenize the dataset and:
+            - ``return_lengths=True`` => add a column with the length of the encoded sequences
+            - ``truncate=True`` => Truncate to the max input length of the model
+        """
+        return tokenizer(batch["sentence1"], batch["sentence2"], return_lengths=True, truncate=True)
+
+    def batch_dataset(batch):
+        """ Pad a batch to the length of the longest sequence in the batch (``padding=True``)
+            And then group the batch in a single example so the sequences will stay together
+        """
+        padded_batch = tokenizer.pad(batch, padding=True)
+        padded_batch = {key: [array] for key, array in padded_batch.items()}
+        return padded_batch
+
+    for split_name, split in glue.items():
+        split = split.map(
+            tokenize, batched=True
+        )  # Tokenize the dataset (adds columns to the dataset w. tokenized inputs)
+        split = split.sort("length")  # Sort the dataset by length
+
+        # Build batches by gathering sequences of similar lengths
+        batch_size = training_args.train_batch_size if split_name == "train" else training_args.eval_batch_size
+        split = split.map(batch_dataset, batched=True, batch_size=batch_size)
+
+        # Set the format of the dataset to output only the columns our model can digest
+        glue[split_name].set_format(columns=["input_ids", "attention_mask", "token_type_ids", "label"])
+
+    # Get the splits
+    train_dataset, eval_dataset, test_dataset = None, None, None
+    if training_args.do_train:
+        train_dataset = glue["train"]
+    if training_args.do_eval:
+        eval_dataset = glue["validation" + ("_matched" if task_name == "mnli" else "")]
+    if training_args.do_predict:
+        test_dataset = glue["test" + ("_matched" if task_name == "mnli" else "")]
+
+    # Define output mode (regression or classification) and num labels (to define the size of the last layer of the model)
+    output_mode = "regression" if "float" in glue["train"].features["label"].dtype else "classification"
+    if output_mode == "regression":
+        num_labels = 1
+    else:
+        num_labels = len(train_dataset.unique("label"))
+
+    if task_name in ["mnli", "mnli-mm"] and tokenizer.__class__.__name__ in (
+        "RobertaTokenizer",
+        "RobertaTokenizerFast",
+        "XLMRobertaTokenizer",
+    ):
+        raise NotImplementedError(
+            """
+        # HACK(label indices are swapped in RoBERTa pretrained model)
+        label_list[1], label_list[2] = label_list[2], label_list[1]
+        """
+        )
```

Comment:
Is there a way to re-order label indices in a `nlp` dataset? Or to override a label_classes like is defined in https://github.com/huggingface/nlp/blob/master/datasets/glue/glue.py#L55 ?

If there was we could just remove those hacky lines and pass the label_classes from the config.json when it has a label2ids dict defined.

Reply:
good point!